src/service.c

```diff
@@ -197,6 +291,18 @@ static void onDispose (GObject *pObject)
             g_free (self->pPrivate->sThemeIcon);
         }
     }
+    else
+    {
+        if (self->pPrivate->nArcticaSubscription)
```

Comment:
I'd love to see the "Arctica" name not be so prominent in the code. GreeterSubscription, maybe?

In fact I have been thinking of renaming Arctica Greeter to Ayatana Greeter several times in the past, so we should not be to Arctica-centric here.

Reply:
Done.